aptos-move/aptos-release-builder/src/components/mod.rs

```diff
@@ -277,6 +277,44 @@ impl ReleaseConfig {
     pub fn parse(serialized: &str) -> Result<Self> {
         serde_yaml::from_str(serialized).map_err(|e| anyhow!("Failed to parse the config: {:?}", e))
     }
+
+    pub fn validate_upgrade(&self, endpoint: Url) -> Result<()> {
+        let client = Client::new(endpoint);
+        if let Some(features) = &self.feature_flags {
+            let on_chain_features = block_on(async {
```

Comment:
noob Rust question - what's the diff between block_on and just doing await?

Reply:
Block_on can only be invoked inside a synchronous function, i.e: `pub fn xxx(...)` whereas async can only be invoked inside an aysnc function `pub async fn xxx(...)`, it is easy to invoke a sync function inside the async function body but not vice versa because async function required a complex runtime to coordinate the execution.